master/internal/api_experiment.go

```diff
@@ -2080,6 +2080,40 @@ func (a *apiServer) GetModelDef(
 	return &apiv1.GetModelDefResponse{B64Tgz: b64Tgz}, nil
 }
 
+func (a *apiServer) GetTaskContextDirectory(
+	ctx context.Context, req *apiv1.GetTaskContextDirectoryRequest,
+) (*apiv1.GetTaskContextDirectoryResponse, error) {
+	if err := a.canDoActionsOnTask(ctx, model.TaskID(req.TaskId),
+		exputil.AuthZProvider.Get().CanGetExperimentArtifacts); err != nil {
+		return nil, err
+	}
+
+	isExp, exp, err := expFromTaskID(ctx, model.TaskID(req.TaskId))
+	if err != nil {
+		return nil, err
+	}
+
+	var tgz []byte
+	if isExp {
+		tgz, err = a.m.db.ExperimentModelDefinitionRaw(exp.ID)
+		if err != nil {
+			return nil, fmt.Errorf(
+				"fetching experiment's taskID %s model definition from database: %w",
+				req.TaskId, err)
+		}
+	} else {
+		tgz, err = db.NonExperimentTasksContextDirectory(ctx, model.TaskID(req.TaskId))
+		if err != nil {
+			return nil, fmt.Errorf(
+				"fetching taskID %s context directory from database: %s", req.TaskId, err)
```

Comment:
nit context in the error; presumably the caller has enough context to know which task ID it was. i've made a call to not include identical info in other calls for this reason, wdyt?

Reply:
As support for including it K8s API will includes it
```
kubectl get node thisnodedontexist
Error from server (NotFound): nodes "thisnodedontexist" not found
```

I'm not sure exactly the reasons why Kubernetes does this or if an API style guide says one or the other. 

I think I mostly have been doing it since it is frustating to call and API and think you have the correct ID but the server tells you not found and you are not sure why.

Reply:
I can buy that. Gotta be careful to add that context only once, though, so you don't end up with "doing thing for abc: load part of abc: xyz abc.."